### PR TITLE
Feature/testing entity classes

### PR DESCRIPTION
**Pull request by danielcapacio**
- added setters and unit tests for entity classes, initial collection tests
- enabling travis-ci, generate and .gitignore code coverage logs
---
I believe the collection tests depend on the list models - not too sure for these.

### CODE REVIEW

**Comment by danielcapacio** [comment]
You can checkout `feature/testing-entity-classes` and run phpunit - all tests should be passing. Going to enable travis-ci now.

File: application/config/autoload.php [hunk 1]
```diff
@@ -132,5 +132,5 @@
 |
 |	$autoload['model'] = array('first_model' => 'first');
 */
-$autoload['model'] = array('flight', 'plane', 'flightslist', 'planeslist');
+$autoload['model'] = array('flight', 'plane', 'flightsList', 'planesList');
```
Comment (nikojohnarellano) [inline application/config/autoload.php:135]: Any reason why these are changed?

Comment (danielcapacio) [reply application/config/autoload.php:135]: Either with the collection models or collection unit test classes - one of the travis builds was producing an error and was trying to access model `$Flightslist`, when it should have been accessing `$FlightsList`.

Comment (danielcapacio) [reply application/config/autoload.php:135]: Which is a cause for ambiguity; unless we re-name our collection classes `Flightslist` / `Planeslist`.

Comment (brynbeaudry) [reply application/config/autoload.php:135]: Code igniter ignores the first capital letter when it autoloads models
So, if you create a class called FlightButt, in order autoload it, you specify you want to autoload 'flightButt'
Based on that, I'd say its okay. If that is the only issue with the PR, it should be approcved. The reason why it was changed was to get it working, and it makes sense that that would make it work, as are class is defined as FlightsList
Therefore we must autoload 'flightsList'. No need to change the classname

**Review by brynbeaudry — APPROVED** [review]
Niko's objections have been met. Merge it in! See my comment

**Pull request merged**